mindsdb/api/http/namespaces/datasource.py

```diff
@@ -117,19 +118,43 @@ def on_file(file):
         return ca.default_store.get_datasource(ds_name)
 
 
+ds_analysis = {}
+
+
+def analyzing_thread(name, default_store):
+    global ds_analysis
+    ds_analysis[name] = None
+    ds = default_store.get_datasource(name)
+    analysis = default_store.get_analysis(ds['name'])
+    ds_analysis[name] = analysis
+
+
 @ns_conf.route('/<name>/analyze')
 @ns_conf.param('name', 'Datasource name')
 class Analyze(Resource):
     @ns_conf.doc('analyse_dataset')
     def get(self, name):
+        global ds_analysis
+        if name in ds_analysis:
+            if ds_analysis[name] is None:
+                return {'status': 'analyzing'}, 200
+            else:
+                analysis = ds_analysis[name]
+                del ds_analysis[name]
+                return analysis, 200
+
         ds = ca.default_store.get_datasource(name)
         if ds is None:
             print('No valid datasource given')
             abort(400, 'No valid datasource given')
 
-        analysis = ca.default_store.get_analysis(ds['name'])
-
-        return analysis, 200
+        if ds['row_count'] <= 10000:
+            analysis = ca.default_store.get_analysis(ds['name'])
+            return analysis, 200
+        else:
+            x = threading.Thread(target=analyzing_thread, args=(name, ca.default_store))
```

Comment:
Does using a thread here actually help ? I'd think it's better to just use a subprocess instead in a similar way we do it for `learn` and `predict`, for now feel free to merge it like this, but if you have time please create an issue around migrating this to a subprocess based approach (via python's multiprocessing, that is).

Reply:
Also, there is currently a "mis usage" of the data analysis in that we refresh the analysis cache to often via `del ds_analysis[name]`. We should consider removing this or adding it behind a time (e.g. only refresh if the analysis is older than 2 days), could you add this functionality as part of this PR ?